packages/flutter_tools/test/commands.shard/hermetic/generate_localizations_test.dart

```diff
@@ -104,6 +104,46 @@ flutter:
     ProcessManager: () => FakeProcessManager.any(),
   });
 
+  testUsingContext('not using synthetic packages (due to --no-implicit-pubspec-resolution)', () async {
+    final Directory l10nDirectory = fileSystem.directory(
+      fileSystem.path.join('lib', 'l10n'),
+    );
+    final File arbFile = l10nDirectory.childFile(
+      'app_en.arb',
+    )..createSync(recursive: true);
+
+    arbFile.writeAsStringSync('''
+{
+  "helloWorld": "Hello, World!",
+  "@helloWorld": {
+    "description": "Sample description"
+  }
+}''');
+    fileSystem.file('pubspec.yaml').writeAsStringSync('''
+flutter:
+  generate: true''');
+
+    final GenerateLocalizationsCommand command = GenerateLocalizationsCommand(
+      fileSystem: fileSystem,
+      logger: logger,
+      artifacts: artifacts,
+      processManager: processManager,
+    );
+    await createTestCommandRunner(command).run(<String>[
+      '--no-implicit-pubspec-resolution',
+      'gen-l10n',
+    ]);
+
+    expect(l10nDirectory.existsSync(), true);
+    expect(l10nDirectory.childFile('app_localizations_en.dart').existsSync(),
+        true);
+    expect(
+        l10nDirectory.childFile('app_localizations.dart').existsSync(), true);
```

Comment:
I think using the `isTrue` matcher instead of `true` would give a slightly better failure message if any of these `expect`s were to fail.

Reply:
I think if anything `isFalse`/`isTrue` will possibly be deprecated in the future: https://github.com/dart-lang/test/issues/2358.